### ISSUE TITLE
Presenter should be generated

### ISSUE THREAD

**Issue by jcoyne** [projecthydra-labs/curation_concerns#667]
### Descriptive summary

When customizing a curationConcerns application, one common task is to customize the presenter.  Thus it would be convenient if that presenter was generated into the host application when the work generator is run, just as is done for the Controller, Actor and Form.

### Expected behavior
When I run `rake curation_concerns:work Book` a class called `BookPresenter` is generated into `app/presenters`  The generated controller (`BooksController`) should be pre-configured to use this presenter.

### Actual behavior
Presenter is not generated.

**Comment by atz** [comment]
You mean `rails generate curation_concerns:work Book` instead of `rake`?

**Comment by jcoyne** [comment]
That's right

**Comment by atz** [comment]
Questions:
- Do we want to refactor the existing presenters that all use `@solr_document` to have a common base class?  Seems like some wins there.
- What is the interaction with the purpose of this generator with CompositePresenterFactory and PresenterFactory, if any?  I don't see a connection, but I also don't fully understand these 2 factories.

**Comment by atz** [comment]
How much do you actually want in this Presenter boilerplate?  Should it just inherit from WorkShowPresenter (or some other parent class) and be otherwise empty?  I would prefer that to copying substantive methods potentially many times and introducing a major maintenance burden or migration blocker.

**Comment by atz** [comment]
Created [generate_presenter](https://github.com/projecthydra/curation_concerns/tree/generate_presenter) branch with initial stuff.

**Comment by jcoyne** [comment]
I don't think we need a base class, we already have this module with common behaviors though:  https://github.com/projecthydra/hydra-editor/blob/master/app/presenters/hydra/presenter.rb

**Comment by jcoyne** [comment]
PresenterFactory instantiates presenters. Primarily for relationships from one presenter to another.

**Comment by atz** [comment]
Nothing else in CC currently uses `Hydra::ActiveModelPresenter` or `Hydra::Presenter`.  I dislike our use of ActiveConcern and the compounding of initialization chains (similar to the hell that is our old `to_solr` logic), especially when more than half the module is deprecated, but I guess it is something.  

Alternatively, CC uses `@solr_document` and ModelProxy instead of @model and Hydra::Presenter, with a level of dynamic indirection here:
https://github.com/projecthydra/curation_concerns/blob/generate_presenter/app/presenters/curation_concerns/model_proxy.rb#L24

That seems preferable to the Hydra::Presenter code, with the caveat that it is odd for methods to depend on a @solr_document that the module includer just has to know to have (and there isn't a base class to handle that for you).
